tensorflow/python/client/timeline.py

```diff
@@ -607,7 +617,48 @@ def _show_memory_counters(self):
                                         total_bytes)
     self._allocator_maximums = alloc_maxes
 
-  def analyze_step_stats(self, show_dataflow=True, show_memory=True):
+  def _preprocess_op_time(self, op_time):
+    """Modified the start and end of ops in step stats"""
+    if op_time == "schedule":
```

Comment:
Do you need the self._origin_step_stats? How about:

if op_time == "schedule":
     return
self._step_stats = copy.deepcopy(self._step_stats)

Reply:
The reason for the self._origin_step_stats is that user may use timeline this way:
```
tl = timeline.Timeline(run_metadata.step_stats)
ctf1 = tl.generate_chrome_trace_format('gpu')
ctf2 = tl.generate_chrome_trace_format()
```
In this case, it would be better to have a origin copy of the step stats.